### PR TITLE
Log when no next block, and add coin type in base scanner

### PR DESCRIPTION
**Pull request by zhiyuan2007**

### CODE REVIEW

File: src/scanner/btc.go [hunk 1]
```diff
@@ -213,7 +216,11 @@ func (s *BTCScanner) waitForNextBlock(block *CommonBlock) (*CommonBlock, error)
 	for {
 		nextBlock, err := s.getNextBlock(block)
 		if err != nil {
-			log.WithError(err).Error("getNextBlock failed")
+			if err.Error() == ErrEmptyBlock.Error() {
```
Comment (gz-c) [inline src/scanner/btc.go]: compare `if err == ErrEmptyBlock`

File: src/scanner/eth.go [hunk 2]
```diff
@@ -112,7 +112,11 @@ func (s *ETHScanner) waitForNextBlock(block *CommonBlock) (*CommonBlock, error)
 	for {
 		nextBlock, err := s.getNextBlock(uint64(block.Height))
 		if err != nil {
-			log.WithError(err).Error("getNextBlock failed")
+			if err.Error() == ErrEmptyBlock.Error() {
```
Comment (gz-c) [inline src/scanner/eth.go]: compare `if err == ErrEmptyBlock`

**Review by gz-c — CHANGES_REQUESTED** [review]

File: src/scanner/btc.go (hunk 1 above)
Comment (zhiyuan2007) [reply src/scanner/btc.go]: no, error can not compare directly , only string can compre

Comment (gz-c) [reply src/scanner/btc.go]: why

Comment (zhiyuan2007) [reply src/scanner/btc.go]: https://stackoverflow.com/questions/39121172/how-to-compare-golang-error-objects

**Review by gz-c — APPROVED** [review]

**Pull request merged**